src/howe_lab_to_nwb/vu2024/utils/add_fiber_photometry.py

```diff
@@ -0,0 +1,293 @@
+from collections import defaultdict
+from copy import deepcopy
+from pathlib import Path
+from typing import Literal, List
+
+import numpy as np
+import pandas as pd
+from neuroconv.tools import get_module
+from neuroconv.utils import calculate_regular_series_rate, FilePathType
+from pynwb import NWBFile
+from ndx_fiber_photometry import (
+    FiberPhotometryTable,
+    OpticalFiber,
+    Indicator,
+    ExcitationSource,
+    Photodetector,
+    DichroicMirror,
+    BandOpticalFilter,
+    FiberPhotometryResponseSeries,
+    FiberPhotometry,
+)
+
+
+def add_photometry_device(nwbfile: NWBFile, device_metadata: dict, device_type: str):
+    """Add a photometry device to an NWBFile object."""
+
+    device_name = device_metadata["name"]
+    if device_name in nwbfile.devices:
+        return
+
+    photometry_device = defaultdict(
+        OpticalFiber=OpticalFiber,
+        Indicator=Indicator,
+        ExcitationSource=ExcitationSource,
+        Photodetector=Photodetector,
+        DichroicMirror=DichroicMirror,
+        BandOpticalFilter=BandOpticalFilter,
+    )[device_type](**device_metadata)
+
+    nwbfile.add_device(photometry_device)
+
+
+def add_fiber_photometry_table(nwbfile: NWBFile, metadata: dict):
+    fiber_photometry_metadata = metadata["Ophys"]["FiberPhotometry"]
+
+    if "FiberPhotometry" in nwbfile.lab_meta_data:
+        return
+
+    fiber_photometry_table_metadata = fiber_photometry_metadata["FiberPhotometryTable"]
+    fiber_photometry_table = FiberPhotometryTable(**fiber_photometry_table_metadata)
+    fiber_photometry_table.add_column(
+        name="allen_atlas_coordinates",
+        description="The fiber bottom coordinates (AP, ML, DV) in Allen Brain Atlas coordinates",
+    )
+    fiber_photometry_table.add_column(
+        name="is_good_fiber",
+        description="Whether the recording from this fiber is of good quality based on post-hoc analysis.",
+    )
```

Comment:
Two additional columns in the `FiberPhotometryTable`:
- `allen_atlas_coordinates` is from `"fiber_top_AP_idx"` `"fiber_top_ML_idx"` `"fiber_top_DV_idx"`  that was mentioned during the meeting, did I get the description right?
- `is_good_fiber`: to annotate the fibers where the "location" could not be determined (noisy fibers) -> do I remember correctly that we decided to include the data from these fibers but annotate whether they are of good quality or not? (Also adding a custom column in the `PlaneSegmentation` table later on)?